cpp/include/cuspatial/soa_readers.hpp

```diff
@@ -16,63 +16,67 @@
 
 #pragma once
 
-typedef struct gdf_column_ gdf_column; // forward declaration
-
 namespace cuspatial {
 
+namespace experimental {
+
 /**
  * @brief read uint32_t data from file as column
  *
  * @param[in] filename: file to read
  *
- * @return gdf_column storing the uint32_t data
+ * @return column storing the uint32_t data
  **/
-gdf_column read_uint32_soa(const char *filename);
+std::unique_ptr<cudf::column> read_uint32_soa(const char *filename, rmm::mr::device_memory_resource* mr);
 
 /**
  * @brief read timestamp data from file as column
  *
  * @param[in] filename: file to read
  *
- * @return gdf_column storing its_timestamp data
+ * @return column storing its_timestamp data
 **/
-gdf_column read_timestamp_soa(const char *filename);
+std::unique_ptr<cudf::column> read_timestamp_soa(const char *filename, rmm::mr::device_memory_resource* mr);
 
 /**
- * @brief read lon/lat from file as two columns; data type is fixed to double (GDF_FLOAT64)
+ * @brief read lon/lat from file as two columns; data type is fixed to FLOAT64
  *
  * @param[in] filename: file name of point data in location_3d layout (lon/lat/alt but alt is omitted)
  *
- * @return gdf_columns storing x and y data
+ * @return columns storing x and y data
 **/
-std::pair<gdf_column, gdf_column>  read_lonlat_points_soa(const char *filename);
+std::pair<std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>>
+read_lonlat_points_soa(const char *filename, rmm::mr::device_memory_resource* mr);
 
 /**
- * @brief read x/y from file as two columns; data type is fixed to double (GDF_FLOAT64)
+ * @brief read x/y from file as two columns; data type is fixed to FLOAT64
  * 
  * @param[in] filename: file name of point data in coordinate_2d layout (x/y)
  * 
- * @return gdf_columns storing x and y data
+ * @return columns storing x and y data
 **/
-std::pair<gdf_column, gdf_column>  read_xy_points_soa(const char *filename);
+std::pair<std::unique_ptr<cudf::column>, std::unique_ptr<cudf::column>>
+read_xy_points_soa(const char *filename, rmm::mr::device_memory_resource* mr);
 
 /**
  * @brief read polygon data from file in SoA format
  * 
- * data type of vertices is fixed to double (GDF_FLOAT64)
+ * data type of vertices is fixed to FLOAT64
  *
  * @param[in] filename: polygon data filename
- * @param[out] ply_fpos: index polygons: prefix sum of number of rings of all
- *             polygons
- * @param[out] ply_rpos: index rings: prefix sum of number of vertices of all
- *             rings
- * @param[out] ply_x: x coordinates of concatenated polygons
- * @param[out] ply_y: y coordinates of concatenated polygons
  *
  * @note: x/y can be lon/lat.
+ *
+ * @return: vector of columns
```

Comment:
Actually, we shouldn't return a std::vector of columns. The columns need to be unique_ptrs. Since there are always 4 of them, we should probably either define a struct, or return a std::tuple.  Vectors are for variable-sized arrays. (Actually, I'm not sure if std::tuple is supported in Cython, need to check.)

```suggestion
 * @return `std::vector` of `cudf::column`s:
```